src/utils/database_util.py

```diff
@@ -184,3 +186,64 @@ def table_weeklychallenge_getlatest(self):
             return None
         sorted_results = sorted(results, key=lambda challenge: challenge["date"])
         return sorted_results[-1]
+
+    def table_weeklychallenge_delete(self, challenge_id: int):
+        """
+        Deletes an item in the Weekly_Challenge database table and 
+        calls weeklyquestion_delete_by_challenge_id() to delete associated questions from
+        Weekly_Question database table
+        """
+        self.table_weeklyquestion_delete_by_challenge_id(challenge_id)
+        table = self.db.table(self.TABLE_WEEKLY_CHALLENGE)
+        results = table.remove(where("id") == challenge_id)
+        return len(results) > 0 #returns false currently as weeklychallenge_insert not implemented
+
+    @validate_insert(required_fields=TABLE_WEEKLY_QUESTION_FIELDS)
+    def table_weeklyquestion_insert(self, item: dict) -> bool:
+        """
+        Inserts an item into the Weekly_Question database table
+        """
+        table = self.db.table(self.TABLE_WEEKLY_QUESTION)
+
+        # Prevent duplicate ID's
+        if len(table.search(where("title_slug") == item["title_slug"])) == 0:
+            table.insert(item)
+            return True
+        return False
+
+    def table_weeklyquestion_load_by_challenge_id(self, challenge_id):
+        """
+        Loads multiple items by challenge id in the Weekly_Question database table
+        """
+        table = self.db.table(self.TABLE_WEEKLY_QUESTION)
+        questions = table.search(where("challenge_id") == challenge_id)
+        if len(questions) == 0:
+            return None
+        return questions
```

Comment:
Consider the return type of this as defined. Technically, it could return a list of questions, or it could return None. Instead of returning None I'd prefer to return an empty list, that way the return type is consistently `list`.

Reply:
Very good point, and makes sense. Updated.